packages/loader/container-loader/src/connectionManager.ts

```diff
@@ -180,7 +180,8 @@ export class ConnectionManager implements IConnectionManager {
 
     private _connectionProps: ITelemetryProperties = {};
 
-    private closed = false;
+    private _closed = false;
+    private _disposed = false;
```

Comment:
It seems like this layer should not care about the difference between dispose() & close().
read-only state is the only difference. I wonder if layers above (Container / DM) where this different makes more sense, should have a layer of "lies" on top of readonly state (similar to how we have a layer of lies RE connection state) to account for this difference. It might be a bit ugly to add it, so maybe it's fine to keep it as is, but it feels like these concepts should not touch this layer.

Reply:
Or maybe it stays here, but we do not transition automatically to readonly state on dispose, instead this layer exposes a method to do so as a separate step and Container layer simply invokes it when needed.
We already have forceReadonly(). Should it be used here by Container? Maybe it has slightly different semantics, not sure.